src/game/SpellEffects.cpp

```diff
@@ -4642,76 +4647,79 @@ void Spell::EffectSummonPet(SpellEffectIndex eff_idx)
-    NewSummon->SetUInt32Value(UNIT_CREATED_BY_SPELL, m_spellInfo->Id);
+        NewSummon->GetCharmInfo()->SetReactState(REACT_DEFENSIVE);
+        NewSummon->SetOwnerGuid(m_caster->GetObjectGuid());
+        NewSummon->SetCreatorGuid(m_caster->GetObjectGuid());
+        NewSummon->setFaction(m_caster->getFaction());
+        NewSummon->SetUInt32Value(UNIT_FIELD_PET_NAME_TIMESTAMP, uint32(time(nullptr)));
+        NewSummon->SetUInt32Value(UNIT_CREATED_BY_SPELL, m_spellInfo->Id);
 
-    NewSummon->InitStatsForLevel(level);
-    NewSummon->InitPetCreateSpells();
+        NewSummon->InitStatsForLevel(level);
+        NewSummon->InitPetCreateSpells();
 
-    map->Add((Creature*)NewSummon);
-    NewSummon->AIM_Initialize();
+        map->Add((Creature*)NewSummon);
+        NewSummon->AIM_Initialize();
 
-    m_caster->SetPet(NewSummon);
-    DEBUG_LOG("New Pet has guid %u", NewSummon->GetGUIDLow());
+        m_caster->SetPet(NewSummon);
+        DEBUG_LOG("New Pet has guid %u", NewSummon->GetGUIDLow());
 
-    if (m_caster->GetTypeId() == TYPEID_PLAYER)
-    {
-        NewSummon->SetUInt32Value(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_NONE);
-        NewSummon->SetUInt32Value(UNIT_FIELD_FLAGS, UNIT_FLAG_PVP_ATTACKABLE);
+        if (m_caster->GetTypeId() == TYPEID_PLAYER)
+        {
+            NewSummon->SetUInt32Value(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_NONE);
+            NewSummon->SetUInt32Value(UNIT_FIELD_FLAGS, UNIT_FLAG_PVP_ATTACKABLE);
 
-        NewSummon->SetByteValue(UNIT_FIELD_BYTES_2, 1, UNIT_BYTE2_FLAG_SUPPORTABLE | UNIT_BYTE2_FLAG_AURAS);
+            NewSummon->SetByteValue(UNIT_FIELD_BYTES_2, 1, UNIT_BYTE2_FLAG_SUPPORTABLE | UNIT_BYTE2_FLAG_AURAS);
 
-        NewSummon->GetCharmInfo()->SetPetNumber(pet_number, true);
+            NewSummon->GetCharmInfo()->SetPetNumber(pet_number, true);
 
-        // generate new name for summon pet
-        NewSummon->SetName(sObjectMgr.GeneratePetName(petentry));
+            // generate new name for summon pet
+            NewSummon->SetName(sObjectMgr.GeneratePetName(petentry));
 
-        if (m_caster->IsPvP())
-            NewSummon->SetPvP(true);
+            if (m_caster->IsPvP())
+                NewSummon->SetPvP(true);
 
-        NewSummon->SavePetToDB(PET_SAVE_AS_CURRENT);
-        ((Player*)m_caster)->PetSpellInitialize();
-    }
-    else
-    {
-        NewSummon->SetUInt32Value(UNIT_NPC_FLAGS, cInfo->NpcFlags);
-        NewSummon->SetUInt32Value(UNIT_FIELD_FLAGS, cInfo->UnitFlags);
-
-        // Notify Summoner
-        if (m_originalCaster && (m_originalCaster != m_caster)
-            && (m_originalCaster->GetTypeId() == TYPEID_UNIT) && ((Creature*)m_originalCaster)->AI())
-            ((Creature*)m_originalCaster)->AI()->JustSummoned(NewSummon);
-        else if ((m_caster->GetTypeId() == TYPEID_UNIT) && ((Creature*)m_caster)->AI())
-            ((Creature*)m_caster)->AI()->JustSummoned(NewSummon);
+            NewSummon->SavePetToDB(PET_SAVE_AS_CURRENT);
```

Comment:
Priest Shadowfiend execution never goes through this code if I am not mistaken, hence the check isnt required here.